synapse/tests/test_lib_stormlib_auth.py

```diff
@@ -411,3 +413,783 @@ async def test_stormlib_auth_default_allow(self):
+            self.true(await core.callStorm('return($lib.user.json.set(hi, $lib.dict(foo=bar, baz=faz)))'))
+            self.true(await core.callStorm('return($lib.user.json.set(bye/bye, $lib.dict(zip=zop, bip=bop)))'))
+            self.eq('bar', await core.callStorm('return($lib.user.json.get(hi, prop=foo))'))
+            self.eq({'foo': 'bar', 'baz': 'faz'}, await core.callStorm('return($lib.user.json.get(hi))'))
+
+            await core.callStorm('$lib.user.json.set(hi, hehe, prop=foo)')
+            items = await core.callStorm('''
+            $list = $lib.list()
+            for $item in $lib.user.json.iter() { $list.append($item) }
+            return($list)
+            ''')
+            self.eq(items, (
+                (('bye', 'bye'), {'zip': 'zop', 'bip': 'bop'}),
+                (('hi',), {'baz': 'faz', 'foo': 'hehe'}),
+            ))
+
+            items = await core.callStorm('''
+            $list = $lib.list()
+            for $item in $lib.user.json.iter(path=bye) { $list.append($item) }
+            return($list)
+            ''')
+            self.eq(items, (
+                (('bye',), {'zip': 'zop', 'bip': 'bop'}),
+            ))
+
+            self.eq('zop', await core.callStorm('return($lib.auth.users.byname(root).json.get(bye/bye, prop=zip))'))
+
+            visi = await core.auth.addUser('visi')
+
+            asvisi = {'user': visi.iden}
+            with self.raises(s_exc.AuthDeny):
+                await core.callStorm('return($lib.auth.users.byname(root).json.get(bye/bye, prop=zip))', opts=asvisi)
+
+            self.none(await core.callStorm('return($lib.user.json.get(hi))', opts=asvisi))
+            await core.callStorm('if (not $lib.user.json.has(hehe)) { $lib.user.json.set(hehe, $lib.dict()) }', opts=asvisi)
+
+            self.true(await core.callStorm('return($lib.user.json.set(hehe, haha, prop=foo))', opts=asvisi))
+            self.true(await core.callStorm('return($lib.user.json.set(hehe, haha, prop=foo))', opts=asvisi))
+            self.eq('haha', await core.callStorm('return($lib.user.json.get(hehe, prop=foo))', opts=asvisi))
+
+            self.eq('haha', await core.callStorm('return($lib.auth.users.byname(visi).json.get(hehe, prop=foo))'))
+            self.true(await core.callStorm('return($lib.auth.users.byname(visi).json.set(hehe, lolz, prop=foo))'))
+            self.eq('lolz', await core.callStorm('return($lib.auth.users.byname(visi).json.get(hehe, prop=foo))'))
+            self.true(await core.callStorm('return($lib.auth.users.byname(visi).json.del(hehe, prop=foo))'))
+            self.none(await core.callStorm('return($lib.auth.users.byname(visi).json.get(hehe, prop=foo))'))
+            self.true(await core.callStorm('return($lib.auth.users.byname(visi).json.del(hehe))'))
+            self.none(await core.callStorm('return($lib.auth.users.byname(visi).json.get(hehe))'))
+            self.false(await core.callStorm('return($lib.auth.users.byname(visi).json.has(hehe))'))
```

Comment:
missing a blank line after here

Reply:
test_stormlib_auth_uservars was duplicated, and now removed.

Reply:
looks like test_stormlib_auth_userjson is duplicated as well